torch/_higher_order_ops/scan.py

```diff
@@ -112,14 +111,9 @@ def add(x: torch.Tensor, y: torch.Tensor):
     # TODO: Support Autograd
     # TODO: Unify handling of pytrees for control flow ops, such as cond, while_loop, etc.
 
-    # Dynamo is expecting a callable with "__code__" attribute.
-    # We cannot directly pass cond_op to it. So we wrap it in a dummy function.
-    def _scan_op_wrapper(*args, **kwargs):
-        return scan(*args, **kwargs)
-
     if not torch._dynamo.is_compiling():
         with _set_compilation_env(), torch._dynamo.utils.disable_cache_limit():
-            return torch.compile(_scan_op_wrapper, backend="eager", fullgraph=True)(
+            return torch.compile(scan, backend="eager", fullgraph=True)(
```

Comment:
We don't need this wrapper because the scan fuction itself is a different frame. We probably should make other control flow operators do the same.

Reply:
This is recursively compiling the scan?

Reply:
yeah, it's like call back to the scan itself if we're not compiling. So the recursion at most one layer.